### PR TITLE
Fixes: #3091 Odd code in CategoryImagesListFragment

### LINKED ISSUE #3091: Odd code in CategoryImagesListFragment
In CategoryImagesListFragment, the code tries to cast the Context to 4 different activities and calls viewPagerNotifyDataSetChanged on each of them. This pattern is present in a few more classes. Does this have any advantages as opposed to implementing an interface?

            try {
                ((CategoryImagesActivity) getContext()).viewPagerNotifyDataSetChanged();
            }catch (Exception e){
                e.printStackTrace();
            }
            try {
                ((CategoryDetailsActivity) getContext()).viewPagerNotifyDataSetChanged();
            }catch (Exception e){
                e.printStackTrace();
            }
            try {
                ((ExploreActivity) getContext()).viewPagerNotifyDataSetChanged();
            }catch (Exception e){
                e.printStackTrace();
            }

### PR DESCRIPTION
**Pull request by rtsketo**
Fixes #3091 Odd code in CategoryImagesListFragment

Created an interface for methods viewPagerNotifyDataSetChanged() and requestMoreImages() commonly used by some Activities.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -6,7 +6,7 @@ buildscript {
         mavenCentral()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.4.1'
+        classpath 'com.android.tools.build:gradle:3.4.2'
```
Comment (ashishkumar468) [inline build.gradle]: Please do this in a separate PR

File: app/src/main/java/fr/free/nrw/commons/RequestAndNotifyInterface.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package fr.free.nrw.commons;
+
+public interface RequestAndNotifyInterface {
```
Comment (ashishkumar468) [inline app/src/main/java/fr/free/nrw/commons/RequestAndNotifyInterface.java]: This could be renamed to something like CategoryImagesCallback, what do you say ?

**Review by ashishkumar468 — CHANGES_REQUESTED** [review]
LGTM :-). Happy to merge after the review comments are addresseD

File: app/src/main/java/fr/free/nrw/commons/RequestAndNotifyInterface.java (hunk 2 above)
Comment (rtsketo) [reply app/src/main/java/fr/free/nrw/commons/RequestAndNotifyInterface.java]: Sounds good! The hardest part for me was to find a name. I'll address your comments ASAP.

**Review by rtsketo — COMMENTED** [review]
Both requested changes were addressed.

**Comment by rtsketo** [comment]
Is there something else I should take care of?

**Comment by ashishkumar468** [comment]
Hi @rtsketo, Thanks fir your PR, I will test the PR and merge it asap.

**Review by neslihanturan — APPROVED** [review]

File: app/src/main/java/fr/free/nrw/commons/bookmarks/pictures/BookmarkPicturesFragment.java [hunk 3]
```diff
@@ -96,11 +97,7 @@ public void onResume() {
             gridView.setVisibility(GONE);
             if (gridAdapter != null) {
                 gridAdapter.clear();
-                try {
-                    ((BookmarksActivity) getContext()).viewPagerNotifyDataSetChanged();
-                }catch (Exception e){
-                    e.printStackTrace();
-                }
+                ((CategoryImagesCallback) getContext()).viewPagerNotifyDataSetChanged();
```
Comment (ashishkumar468) [inline app/src/main/java/fr/free/nrw/commons/bookmarks/pictures/BookmarkPicturesFragment.java]: Instead of  casting the context to CategoryImagesCallback, should we consider actually passing the callback to the fragment ?

File: app/src/main/java/fr/free/nrw/commons/category/CategoryImagesCallback.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package fr.free.nrw.commons.category;
```
Comment (ashishkumar468) [inline app/src/main/java/fr/free/nrw/commons/category/CategoryImagesCallback.java]: Javadocs please

**Review by ashishkumar468 — CHANGES_REQUESTED** [review]
Happy to merge after the comments are addressed :)

File: app/src/main/java/fr/free/nrw/commons/bookmarks/pictures/BookmarkPicturesFragment.java (hunk 3 above)
Comment (rtsketo) [reply app/src/main/java/fr/free/nrw/commons/bookmarks/pictures/BookmarkPicturesFragment.java]: That's correct, my mistake.

**Comment by rtsketo** [comment]
@ashishkumar468 , is there anything else in need to be changed?

**Review by neslihanturan — APPROVED** [review]
Thanks @rtsketo :)

**Comment by neslihanturan** [comment]
I see that @ashishkumar468 's reviews are also fixed, so merging it :)

**Review by maskaravivek — APPROVED** [review]

**Review by domdomegg — APPROVED** [review]
